cpp/src/io/json/json_column.cu

```diff
@@ -826,10 +961,17 @@ table_with_metadata device_parse_nested_json(device_span<SymbolT const> d_input,
 
   // Verify that we were in fact given a list of structs (or in JSON speech: an array of objects)
   auto constexpr single_child_col_count = 1;
-  CUDF_EXPECTS(data_root.type == json_col_t::ListColumn and
-                 data_root.child_columns.size() == single_child_col_count and
-                 data_root.child_columns.begin()->second.type == json_col_t::StructColumn,
-               "Currently the nested JSON parser only supports an array of (nested) objects");
+  if (is_array_of_arrays) {
+    CUDF_EXPECTS(data_root.type == json_col_t::ListColumn and
+                   data_root.child_columns.size() == single_child_col_count and
+                   data_root.child_columns.begin()->second.type == json_col_t::ListColumn,
+                 "Currently the nested JSON parser only supports an array of arrays");
+  } else {
+    CUDF_EXPECTS(data_root.type == json_col_t::ListColumn and
+                   data_root.child_columns.size() == single_child_col_count and
+                   data_root.child_columns.begin()->second.type == json_col_t::StructColumn,
+                 "Currently the nested JSON parser only supports an array of (nested) objects");
+  }
```

Comment:
These error messages seem incorrect. Looks like this code check for invalid input. My take (please excuse the formatting):
```suggestion
  CUDF_EXPECTS(data_root.type == json_col_t::ListColumn and
                 data_root.child_columns.size() == single_child_col_count and
                 data_root.child_columns.begin()->second.type == is_array_of_arrays ?
                   json_col_t::ListColumn :
                   json_col_t::StructColumn,
                 "Input needs to be an array of arrays or an array of (nested) objects");
```